src/inspector/inspector_socket_server.h

```diff
@@ -16,24 +16,39 @@ class InspectorSocketServer;
 class SocketSession;
 class ServerSocket;
 
-class SocketServerDelegate {
-public:
-  virtual void AssignServer(InspectorSocketServer* server) = 0;
-  virtual void StartSession(int session_id, const std::string& target_id) = 0;
-  virtual void EndSession(int session_id) = 0;
-  virtual void MessageReceived(int session_id, const std::string& message) = 0;
-  virtual std::vector<std::string> GetTargetIds() = 0;
-  virtual std::string GetTargetTitle(const std::string& id) = 0;
-  virtual std::string GetTargetUrl(const std::string& id) = 0;
-  virtual ~SocketServerDelegate() {}
+class InspectorAgentDelegate {
+ public:
+  InspectorAgentDelegate(
+      AgentImpl& agent,
+      const std::string &script_path,

```

Comment:
> & [](start = 23, length = 2)

(nit) could these be made && to avoid copying?

Reply:
It looks like the copies are needed due to threading and lifetimes.

---
In reply to: [447996815](https://github.com/microsoft/v8-jsi/pull/17#discussion_r447996815) [](ancestors = 447996815)